test/test_document.rb

```diff
@@ -173,6 +173,18 @@ def test_empty_value
             REXML::Document.new(xml)
           end
         end
+
+        def test_empty_entity
+          xml = <<EOF
+<!DOCTYPE root [
+  <!ENTITY>
+]>
+EOF
+
+          assert_raise(REXML::ParseException) do
+            REXML::Document.new(xml)
+          end
+        end
```

Comment:
Could you create a new file as `test/parse/test_entity_declaration.rb` and use the same style as https://github.com/ruby/rexml/blob/master/test/parse/test_notation_declaration.rb ?

e.g.:

```ruby
# frozen_string_literal: false
require 'test/unit'
require 'rexml/document'

module REXMLTests
  class TestParseEntityDeclaration < Test::Unit::TestCase
    private
    def xml(internal_subset)
      <<-XML
<!DOCTYPE r SYSTEM "urn:x-henrikmartensson:test" [
#{internal_subset}
]>
<r/>
      XML
    end

    def parse(internal_subset)
      REXML::Document.new(xml(internal_subset)).doctype
    end

    def test_empty
      exception = assert_raise(REXML::ParseException) do
        parse(<<-INTERNAL_SUBSET)
<!ENTITY>
        INTERNAL_SUBSET
       end
        assert_equal(<<-DETAIL.chomp, exception.to_s)
Malformed notation declaration: name is missing
Line: 5
Position: 72
Last 80 unconsumed characters:
 <!ENTITY>  ]> <r/> 
        DETAIL
      end
    end
  end
end
```

Reply:
test/parse/test_entity_declaration.rb was created.